packages/samples/headless-react/src/components/smart-snippet/smart-snippet.class.tsx

```diff
@@ -90,6 +91,23 @@ export class SmartSnippet extends Component<{}, SmartSnippetState> {
             >
               Thumbs down
             </button>
+            {source ? (
+              <a
+                href={filterProtocol(source.clickUri)}
+                onClick={() => this.controller.source?.select()}
+                onContextMenu={() => this.controller.source?.select()}
+                onMouseDown={() => this.controller.source?.select()}
+                onMouseUp={() => this.controller.source?.select()}
+                onTouchStart={() =>
+                  this.controller.source?.beginDelayedSelect()
+                }
+                onTouchEnd={() => this.controller.source?.cancelPendingSelect()}
+              >
+                Source
+              </a>
+            ) : (
+              []
```

Comment:
how come you return an empty array here?
isn't it cleaned to do 
`Boolean(source) && <Component />` ?

Reply:
in a few instances, I saw `false` being rendered. So now, I'm avoiding using the `&&` pattern in JSX.

Reply:
odd! I've seen 0 before, but never false.
fair enough

Reply:
I may have seen it in attributes

Reply:
I personally prefer extracting any logic from the JSX code, and just have a `renderSource()` function which itself contains the `if/else` code.

Just seems easier to read ?

Reply:
In Atomic that's generally what I do, I didn't do it here for the sake of having the whole thing in the same function, but I'll extract it.